src/test/java/com/amplitude/api/AmplitudeClientTest.java

```diff
@@ -2130,4 +2131,235 @@ public void run(MiddlewarePayload payload, MiddlewareNext next) {
+        assertEquals((long)dbHelper.getLongValue(AmplitudeClient.LAST_EVENT_TIME_KEY), timestamps[3]);
+
+        looper.runToEndOfTasks();
+        looper.runToEndOfTasks();
+        RecordedRequest request = runRequest(amplitude);
+        JSONArray events = getEventsFromRequest(request);
+        assertEquals(events.length(), 1);
+        JSONObject identify = events.getJSONObject(0);
+        assertEquals(identify.getString("event_type"), Constants.IDENTIFY_EVENT);
+        assertEquals(identify.getLong("event_id"), 1);
+        assertEquals(identify.getLong("timestamp"), timestamps[0]);
+        assertEquals(identify.getLong("sequence_number"), 1);
+        JSONObject userProperties = identify.getJSONObject("user_properties");
+        assertEquals(userProperties.length(), 1);
+        assertTrue(userProperties.has(Constants.AMP_OP_SET));
+        JSONObject expected = new JSONObject();
+        expected.put("key1", "key1-value2");
+        expected.put("key2", "key2-value2");
+        expected.put("key3", "key3-value2");
+        expected.put("key4", "key4-value2");
+        assertTrue(Utils.compareJSONObjects(userProperties.getJSONObject(Constants.AMP_OP_SET), expected));
+        assertEquals(0, getIdentifyInterceptorCount());
+        assertEquals((long)dbHelper.getLastIdentifyInterceptorId(), -1L);
+        assertEquals((long)dbHelper.getLongValue(AmplitudeClient.LAST_IDENTIFY_ID_KEY), 1L);
+    }
+
+    @Test
+    public void testMultipleIdentifyWithSetActionOnlyAndOneEvent() throws JSONException {
+        long [] timestamps = {1000, 1001, 1002, 1003, 1004};
+        clock.setTimestamps(timestamps);
+        ShadowLooper looper = Shadows.shadowOf(amplitude.logThread.getLooper());
+        looper.runToEndOfTasks();
+        Middleware middleware = new Middleware() {
+            @Override
+            public void run(MiddlewarePayload payload, MiddlewareNext next) {
+                try {
+                    if (!payload.event.getString("event_type").equals(Constants.IDENTIFY_EVENT)) {
+                        payload.event.optJSONObject("user_properties").put("key1", "key1-value3")
+                                .put("key2", "key2-value3");
+                    }
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+
+                next.run(payload);
+            }
+        };
+        amplitude.addEventMiddleware(middleware);
+        amplitude.identify(new Identify().set("key1", "key1-value1").set("key2", "key2-value1").set("key3", "key3-value1"));
+        amplitude.identify(new Identify().set("key1", "key1-value2").set("key4", "key4-value1"));
+        amplitude.identify(new Identify().set("key2", "key2-value2"));
+        amplitude.identify(new Identify().set("key3", "key3-value2").set("key4", "key4-value2"));
+        amplitude.logEvent("test_event", new JSONObject().put("test_event_prop_key", "test_event_prop_value"));
```

Comment:
NIT: One minor case to check is getting 4 identifies, 1 event and 1 identify. This should translate to 1 event and 1 identify.

Reply:
@izaaz Sounds good, I just add a unit test for the described case.